### PR TITLE
feat: fix input on landing page

### LINKED ISSUE #178: Refactor the asset selection and error UI
The asset selection and related error handling are inconsistent which result in many logical fallacies. Design a better API so this can be scaled well. Explore cache management and gauge if this is triggering these issues. Under the current implementation, multiple instances of logical fallacy are experienced while selecting an asset and entering the amount. 

1. Go to Safari on iOS and enter the Boltz URL
2. Select on-chain LTC as base and quote pair 
3. Delete the default amount from the amount-entry field
4. Change the receive-asset to lightning BTC
5. The amount-entry field is disabled and "Coming soon" error gets thrown

All the permutations and combinations of asset base and quote pairs along with the related two errors should be consistent and retain the context of the previous state. Better cache management is desirable. 

 - Device: iPhone XR
 - OS: iOS [12.2]
 - Browser: Safari
 - Browser Version: iOS 12.2 stock safari

### LINKED ISSUE #182: Not supported pairs do not show an error
**Describe the bug**
Unsupported pairs do not show the expected error message:
image

**To Reproduce**
Select a unsupported pair like LTC/BTC or lightning LTC/lightning BTC

**Expected behavior**
Show an error message if the pair is not supported

### PR DESCRIPTION
**Pull request by ImmanuelSegol**
@michael1011 This pr is supposed to solve the rounding issues please test it.

closes #182
closes #178

### CODE REVIEW

File: src/components/swaptab/mobileswaptab.js [hunk 1]
```diff
@@ -48,10 +48,10 @@ const MobileSwapTabContent = ({
             className={classes.inputMobile}
             min={minAmount}
             max={maxAmount}
-            step={0.001}
+            step={qouteStep}
```
Comment (michael1011) [inline src/components/swaptab/mobileswaptab.js]: Why is this `quoteStep` (btw: typo `quote`)? This is the base amount.

File: src/components/swaptab/desktopswaptab.js [hunk 2]
```diff
@@ -182,26 +179,27 @@ DeskTopSwapTabContent.propTypes = {
...
   disabled: PropTypes.bool,
-  parseRate: PropTypes.func,
+  rate: PropTypes.string,
   switchPair: PropTypes.func,
   updateQuoteAmount: PropTypes.func,
   updateBaseAmount: PropTypes.func,
   updatePair: PropTypes.func,
   shouldSubmit: PropTypes.func,
+  baseStep: PropTypes.string,
+  qouteStep: PropTypes.any,
```
Comment (michael1011) [inline src/components/swaptab/desktopswaptab.js]: Both, `baseStep` and `quoteStep`, should have the same type.

File: src/components/swaptab/swaptabwrapper.js [hunk 3]
```diff
@@ -232,43 +280,64 @@ class SwapTabWrapper extends React.Component {
...
-    });
+    if (this.state.rate) {
+      this.setState(
+        {
+          base: this.state.quote,
+          quote: this.state.base,
+        },
+        () => {
+          const symbol = this.getSymbol();
+          const limits = this.props.limits[symbol].minimal / decimals;
```
Comment (michael1011) [inline src/components/swaptab/swaptabwrapper.js]: Why use the minimal limit as default after switching the assets? Getting the quote amount, setting it as new base amount and `updateQuoteAmount` should work fine.

**Comment by michael1011** [comment]
https://github.com/BoltzExchange/boltz-frontend/pull/177#discussion_r281795508 is still not fixed

**Comment by michael1011** [comment]
>> The step of the "You send" amount behaves weird: type 0.001 into the text field -> step up -> amount is now 0.0017874

> I think this is a JS floating point issue.

No that is really unlikely because `0.0007` would be a little big for a floating point issue, wouldn't it? Could you have another look at that code? Maybe there is another cause

**Comment by ImmanuelSegol** [comment]
> > > The step of the "You send" amount behaves weird: type 0.001 into the text field -> step up -> amount is now 0.0017874
> 
> > I think this is a JS floating point issue.
> 
> No that is really unlikely because `0.0007` would be a little big for a floating point issue, wouldn't it? Could you have another look at that code? Maybe there is another cause

I think it's fixed now.

**Pull request merged**